File: Nov2020/12Nov/PolygonPossibility.cpp
/*
Problem name: Polygon Possibility
Problem link: hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/polygon-possible/
*/

#include <bits/stdc++.h>
using namespace std;

int main(){
    int t;
	long int n;
	cin>>t;
	while(t--){
		cin>>n;
		long int l[n],sum=0,flag=0;
		for(int i=0;i<n;++i){
			cin>>l[i];
			sum+=l[i];
		}	
		for(int i=0;i<n;++i){
			if(l[i]>=sum-l[i]){
				cout<<"No"<<endl;
				flag=1;
				break;
			}
		}
		if(flag==0)	cout<<"Yes"<<endl;
	}
    return 0;
}

File: Nov2020/12Nov/PolygonPossibility.java
/*
Problem Name: Polygon Possibility
Problem link: https://www.hackerearth.com/practice/data-structures/arrays/1-d/practice-problems/algorithm/polygon-possible/
*/
import java.util.*;
class PolygonPossibility {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(;t>0;t--){
            int n = in.nextInt();
            int arr[] = new int[n];
            for(int i=0;i<n;i++)
                arr[i] = in.nextInt();
            Arrays.sort(arr);
            for(int i=1;i<n-1;i++)
                arr[i] += arr[i-1];
            if(arr[n-1] < arr[n-2])
                System.out.println("Yes");
            else
                System.out.println("No");
        }
    }
}
